plugins/user-authenticators/ldap/src/main/java/org/apache/cloudstack/api/command/LdapListUsersCmd.java

```diff
@@ -66,34 +121,113 @@ public LdapListUsersCmd(final LdapManager ldapManager, final QueryService queryS
         _queryService = queryService;
     }
 
+    /**
+     * (as a check for isACloudstackUser is done) only non cloudstack users should be shown
+     * @param users a list of {@code LdapUser}s
+     * @return a (filtered?) list of user response objects
+     */
     private List<LdapUserResponse> createLdapUserResponse(final List<LdapUser> users) {
         final List<LdapUserResponse> ldapResponses = new ArrayList<LdapUserResponse>();
         for (final LdapUser user : users) {
-            if (getListType().equals("all") || !isACloudstackUser(user)) {
-                final LdapUserResponse ldapResponse = _ldapManager.createLdapUserResponse(user);
-                ldapResponse.setObjectName("LdapUser");
-                ldapResponses.add(ldapResponse);
-            }
+            final LdapUserResponse ldapResponse = _ldapManager.createLdapUserResponse(user);
+            ldapResponse.setObjectName("LdapUser");
+            ldapResponses.add(ldapResponse);
         }
         return ldapResponses;
     }
 
+    private List<UserResponse> cloudstackUsers = null;
+
     @Override
     public void execute() throws ServerApiException {
-        List<LdapUserResponse> ldapResponses = null;
+        List<LdapUserResponse> ldapResponses = new ArrayList<LdapUserResponse>();
         final ListResponse<LdapUserResponse> response = new ListResponse<LdapUserResponse>();
         try {
-            final List<LdapUser> users = _ldapManager.getUsers(null);
+            final List<LdapUser> users = _ldapManager.getUsers(domainId);
             ldapResponses = createLdapUserResponse(users);
+//            now filter and annotate
+            ldapResponses = applyUserFilter(ldapResponses);
         } catch (final NoLdapUserMatchingQueryException ex) {
-            ldapResponses = new ArrayList<LdapUserResponse>();
+            // ok, we'll make do with the empty list ldapResponses = new ArrayList<LdapUserResponse>();
         } finally {
             response.setResponses(ldapResponses);
             response.setResponseName(getCommandName());
             setResponseObject(response);
         }
     }
 
+    private List<UserResponse> getCloudstackUsers() {
+        // get a list of relevant cloudstack users, meaning
+        // if we are filtering for local domain, only get users for the current domain
+        // if we are filtering for any domain, get recursive all users for the root domain
+        // if we are filtering for potential imports,
+        //    we are only looking for users in the linked domains/accounts,
+        //    which is only relevant if we ask ldap users for this domain.
+        //    So we are asking for all users in the current domain as well
+        // in case of no filter we should find all users in the current domain for annotation.
+        if (cloudstackUsers == null) {
+            ListResponse<UserResponse> cloudstackUsersresponse;
+            switch (getUserFilter()) {
```

Comment:
Alternative implementation idea: Make getUserFilter return an enum

Reply:
I investigated this and it is possible but quite ellaborate. the issue there is that this creates a static context in the enum while several services and a instance variable list of cloudstack users are being used in the filter methods. It is certainly possible but do we think it is worth the effort, @rhtyd ?